packages/flutter/test/material/ink_well_test.dart

```diff
@@ -1330,4 +1333,253 @@ void main() {
+
+      // Does not call setState.
+      await tester.pumpWidget(buildApp(GlobalKey()));
+      await tester.tap(find.byType(InkWell));
+      await tester.pump(const Duration(milliseconds: 60));
+      expectTest(true);
+      await tester.pumpAndSettle();
+      expectTest(false);
+
+      await tester.pumpWidget(buildApp(GlobalKey(), onTapChangeWrap: true));
+      await tester.tap(find.byType(InkWell));
+      await tester.pump(const Duration(milliseconds: 60));
+      expectTest(true);
+      await tester.pumpAndSettle();
+      expectTest(false);
+
+      await tester.pumpWidget(buildApp(GlobalKey(), onTapDownChangeWrap: true));
+      final TestGesture testGesture = await tester.startGesture(tester.getRect(find.byType(InkWell)).center);
+      await tester.pump(const Duration(milliseconds: 60));
+      expectTest(true);
+      await testGesture.up();
+      await tester.pumpAndSettle();
+      expectTest(false);
+    }
+
+    // InkWell have a key
+    await test((Key? key, {bool onTapDownChangeWrap = false, bool onTapChangeWrap = false}) {
+      bool wrap = false;
+
+      return MaterialApp(
+        home: Scaffold(
+          body: Center(
+            child: StatefulBuilder(
+              builder: (BuildContext context, void Function(void Function()) setState) {
+                Future<void> changeWrap() async {
+                  await Future<void>.delayed(const Duration(milliseconds: 50));
+
+                  setState(() {
+                    wrap = !wrap;
+                  });
+                }
+
+                Widget child = InkWell(
+                  key: key,
+                  onTapDown: (_) {
+                    if (onTapDownChangeWrap)
+                      changeWrap();
+                  },
+                  onTap: () {
+                    if (onTapChangeWrap)
+                      changeWrap();
+                  },
+                  child: Container(
+                    height: 160,
+                    alignment: Alignment.center,
+                  ),
+                );
+
+                if (wrap) {
+                  child = Container(
+                    margin: const EdgeInsets.only(top: 320),
+                    child: child,
+                  );
+                }
+
+                return child;
+              },
+            ),
+          ),
+        ),
+      );
+    });
+
+    // Ancestor widget have Key
+    await test((Key? key, {bool onTapDownChangeWrap = false, bool onTapChangeWrap = false}) {
+      bool wrap = false;
+
+      return MaterialApp(
+        home: Scaffold(
+          body: Center(
+            child: StatefulBuilder(
+              builder: (BuildContext context, void Function(void Function()) setState) {
+                Future<void> changeWrap() async {
+                  await Future<void>.delayed(const Duration(milliseconds: 50));
+
+                  setState(() {
+                    wrap = !wrap;
+                  });
+                }
+
```

Comment:
these blank lines are mildly confusing. Blank lines read like paragraph breaks, but here you have one inside a nested closure and then outside that closure in the same statement, so it's not clear what is going on. I would just remove all the blank lines and let the indentation speak for itself.